app/services/authentication_service.rb

```diff
@@ -0,0 +1,16 @@
+class AuthenticationService
+  def initialize(api_keys)
+    @api_keys = api_keys
+  end
+
+  def api_key_or_nil(api_key)
+    @api_keys.find_api_key(api_key)
+  end
+
+  # :reek:UtilityFunction
+  def private_api_key_valid?(api_key, private_key)
+    return false unless api_key
```

Comment:
we could extend the validate_api_key_hash method, so it raises an error if their exists an invalid api_key, what do you think?